runtime/j9vm/java11vmi.c

```diff
@@ -213,13 +216,17 @@ freePackage(J9VMThread *currentThread, J9Package *j9package)
 			hashTableFree(j9package->exportsHashTable);
 		}
 #if defined(J9VM_OPT_SNAPSHOTS)
-		if (IS_SNAPSHOTTING_ENABLED(vm)) {
+		if (IS_SNAPSHOTTING_ENABLED(vm) && (NULL != j9package->packageName)) {
 			VMSNAPSHOTIMPLPORT_ACCESS_FROM_JAVAVM(vm);
 			vmsnapshot_free_memory((void *)j9package->packageName);
+			j9package->packageName = NULL;
 		} else
 #endif /* defined(J9VM_OPT_SNAPSHOTS) */
 		{
-			j9mem_free_memory((void *)j9package->packageName);
+			if (NULL != j9package->packageName) {
+				j9mem_free_memory((void *)j9package->packageName);
+				j9package->packageName = NULL;
+			}
 		}
```

Comment:
`packageName` can be set null from a single place after the above if/else block since both blocks `free` the memory for `packageName`.
```suggestion
		}
		j9package->packageName = NULL;
```